templates/gitlab/mender-deployment-create.gitlab-ci.yml

```
@@ -0,0 +1,50 @@
+# Use this template to create deployments for Mender artifacts from MENDER_ARTIFACTS_FOLDER folder.
+#
+# The template is designed to be used with the includes keyword or copied into an existing
+# configuration. It requires `MENDER_SERVER_URL` and `MENDER_SERVER_ACCESS_TOKEN` variables
+# to work and can be used to create a deployment on a Mender server.
+#
+# It's possible to set custom values by setting the following environment variables:
+# - `MENDER_DEPLOYMENT_GROUP`: The name of Mender devices group a deployment will be triggered to.
+#                              One of `MENDER_DEPLOYMENT_GROUP` or `MENDER_DEVICES_LIST` is required.
+#                              `MENDER_DEVICES_LIST` takes the priority if both are set.
+# - `MENDER_DEVICES_LIST`:     The list of Mender devices a deployment will be triggered to.
+# - `MENDER_DEPLOYMENT_NAME`:  The name of Mender deployment.
+# - `MENDER_RELEASE_NAME`:     The name of Mender release.
+
+include:
+  - project: 'Northern.tech/Mender/mender-ci-workflows'
+    file: 'templates/gitlab/common.gitlab-ci.yml'
+    ref: 'master'
+
+variables:
+  MENDER_DEPLOYMENT_GROUP: "" # The name of Mender devices group a deployment will be triggered to. One of `MENDER_DEPLOYMENT_GROUP` or `MENDER_DEVICES_LIST` is required. `MENDER_DEVICES_LIST` takes the priority if both are set
+  MENDER_DEVICES_LIST: ""     # The list of Mender devices a deployment will be triggered to
+  MENDER_RELEASE_NAME: ""     # The name of Mender release
+  MENDER_DEPLOYMENT_NAME: ""  # The name of Mender deployment
+
+.mender:create:deployment:
+  image: mendersoftware/mender-ci-tools:mender-master
+  script:
```

(related to the above comment), add here an early check for the requirements
```suggestion
  script:
    - if [ -z "${MENDER_DEPLOYMENT_NAME}"; then
      -   'echo "ERROR: MENDER_DEPLOYMENT_NAME is not defined found."'
      -   exit 1
    - fi
```

Are there other dependencies?

---

Added proper input validation